### DISCUSSION TITLE
Result combination from load case or result combination

### DISCUSSION THREAD

**Discussion by DannieCW** [Dlubal-Software/RFEM_Python_Client#282] [Q&A]
Dear developers,

I am trying to create result combinations via API/web services. I have a question:

In result combination no. 5, I am trying to combine a few load cases 1, 2, 3, 4, 5 
In result combination no. 4, I am trying to combine existing result combinations 1, 2, 3

Printing the 2 objects gives the following, where I do not immediately see the difference where load cases or result combinations are used, except the name of the result combination.
Could you help to demonstrate how to differentiate the result combination and load case to create an additional result combination?

(result_combination){
   no = 5
   design_situation = 1
   user_defined_name_enabled = False
   name = "LC1 + LC2 + LC3 or to LC5"
   to_solve = True
   comment = None
   combination_type = "COMBINATION_TYPE_GENERAL"   
   srss_combination = False
   items = 
      (array_of_result_combination_items){
         result_combination_items[] = 
            (result_combination_items_row){        
               no = 1
               description = None
               row = 
                  (result_combination_items){      
                     case_object_item = 1
                     operator_type = "OPERATOR_AND"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 2
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 2
                     operator_type = "OPERATOR_AND"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 3
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 3
                     operator_type = "OPERATOR_OR"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 4
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 4
                     operator_type = "OPERATOR_OR"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 5
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 5
                     operator_type = "OPERATOR_NONE"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
      }
   generate_subcombinations = False
   is_generated = False
   consider_construction_stage_active = False
   id_for_export_import = "1W_xTudkL36wePETcmf9zE"
   metadata_for_export_import = None
 }
(result_combination){
   no = 4
   design_situation = 1
   user_defined_name_enabled = False
   name = "RC1 + RC2 + RC3"
   to_solve = True
   comment = None
   combination_type = "COMBINATION_TYPE_GENERAL"
   srss_combination = False
   items =
      (array_of_result_combination_items){
         result_combination_items[] =
            (result_combination_items_row){
               no = 1
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 1
                     operator_type = "OPERATOR_AND"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 2
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 2
                     operator_type = "OPERATOR_AND"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
            (result_combination_items_row){
               no = 3
               description = None
               row =
                  (result_combination_items){
                     case_object_item = 3
                     operator_type = "OPERATOR_NONE"
                     left_parenthesis = False
                     right_parenthesis = False
                     case_object_factor = 1.0
                     case_object_load_type = "LOAD_TYPE_TRANSIENT"
                  }
            },
      }
   generate_subcombinations = False
   is_generated = False
   consider_construction_stage_active = False
   id_for_export_import = "0WUiL7aHb3oxe0gb92iTeT"
   metadata_for_export_import = None
 }

**Comment by aliciaknauer** [comment]
Dear Dannie!

Thank you for your question, I am happy to help!

In the round brackets you have the function and in the curly brackets are its attributes. 
You are correct, one criteria to differentiate between the two is trough the name variable. 
If you see variables like row in your print, in most cases this implies a table. If you scroll down in your output you can see that for the first result combination there are 5 rows which you can identify by (result_combinations_item_row), which makes sense since you want the load cases 1 to 5 in it. Each row has again variable no from 1-5
You can also identify you first result combination by the variable no=5.  

Now, if you scroll down till the next (result_combination) you can find underneath directly no=4. This means you found your second result combination.
If you look again for the (result_combinations_items) you can identify the three rows underneath, which makes sense since this result combination is made up of RC1, RC2, RC3.

I attached a picture which I hope clears everything up and hopefully I was able to help you!

If you have any more questions, please don't hesitate and ask away!

Kind regards, Alicia


![result_combinations](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/74434856/e31e63bf-ee56-4278-a04c-15d0db71521e)

**Reply by DannieCW** [reply]
Hi Alicia, 

Thanks for the detailed explanation with the picture demonstration. 

What I meant was in the picture for example:  the RC4 and RC5 consist of RC1-3 and LC 1- 5 respectively. 
In the code, both indicate no = 1, 2 , 3 and no = 1, 2, 3, 4, 5

Is there somewhere I can specify if no =1 is the Load case LC1 or the Result combination RC1? 

![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/138588349/f5c3c2e3-b424-4240-ada3-a4db9ecf84ac)

Best regards,
Dannie

**Comment by aliciaknauer** [comment]
Dear Danni!

Unfortunately right now it is not possible. I tried to find a way around it and found out that it is indeed a bug. But we are working on it and when we have solved it, we will let you know immediately. 
Thank you for bringing it up! 

Kind regards, Alicia

**Reply by Dervis-vL** [reply]
@aliciaknauer do you possibly have an indicaation how far along the progress to solving this bug would be?

Kind regards,

**Reply by aliciaknauer** [reply]
Hi @Dervis-vL,

unfortunately it is not solved yet, but the team is working on it. When it is solved I will let you know immediately. 
Thank you for your patience!
Kind regards, Alicia
